auth.go

```diff
@@ -85,22 +107,28 @@ func prepareUsernamePasswordLogin(username string, password string) (string, str
 	}
 }
 
-func prepareServiceAccountLogin(username, password string) (string, string, string) {
+// prepareServiceAccountLogin returns the loginUrl, contentType and body needed to query an access token using a client id and a client secret
+func prepareServiceAccountLogin(clientId, clientSecret string) (string, string, string) {
 	var authUrl = ServiceAccountLoginURLSuffix
 	form := url.Values{}
-	form.Add("client_id", username)
-	form.Add("client_secret", password)
+	form.Add("client_id", clientId)
+	form.Add("client_secret", clientSecret)
 	form.Add("grant_type", "client_credentials")
 	var body = form.Encode()
 	return authUrl, ContentTypeForm, body
 }
 
 // deleteAccessTokenFromCache deletes an access token from the cache if available
 func deleteAccessTokenFromCache(username, apiEndpoint string) {
-	cache.Delete(getCacheKey(username, apiEndpoint))
+	if tokenCache != nil {
+		err := tokenCache.Delete(getCacheKey(username, apiEndpoint))
+		if err != nil {
+			infolog.Println(fmt.Errorf("could not remove token from the memory cache: %v", err))
+		}
+	}
 }
 
-// isServiceAccount checks if a username is a service account cliend id
+// isServiceAccount checks if a username is a service account client id
 func isServiceAccount(username string) bool {
-	return strings.Contains(username, "@")
+	return !strings.Contains(username, "@")
```

Comment:
well spotted, do we have an integration test for service account?

Reply:
Yes we have tests for SA. It was not an bug, the function name was just wrong (should have been "isNotServiceAccount") 
```
                 if isServiceAccount(username) {
			loginUrl, contentType, body, err = prepareUsernamePasswordLogin(username, password)
		} else {
		        loginUrl, contentType, body = prepareServiceAccountLogin(username, password)
		}
```		
Since it's better to have affirmative names I changed the implementation of the function and the if statement

```
		if isServiceAccount(username) {
			loginUrl, contentType, body = prepareServiceAccountLogin(username, password)
		} else {
			loginUrl, contentType, body, err = prepareUsernamePasswordLogin(username, password)
			...
		}
```